src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -63,42 +63,26 @@ public EmailProcessor() {
      */
     public CompletionStage<Statistics> processEmails(final TenantConfiguration tenantConfig) {
         SphereClient client = tenantConfig.getSphereClient();
-        CustomObjectQuery<JsonNode> query = CustomObjectQuery.ofJsonNode().byContainer(CONTAINER_ID).withLimit(100L)
-            .withSort(s -> s.createdAt().sort().asc());
-        return client
-            .execute(query)
-            .thenApply(response -> {
-                Statistics statistics = new Statistics(tenantConfig.getProjectKey());
-                if (response.getTotal() < 1) {
-                    LOG.info(String.format("No email to process for tenant %s", tenantConfig
-                        .getProjectKey()));
-                }
-                List<CompletableFuture<Void>> allTenants = response.getResults().stream()
-                    .map(customObject -> Optional.ofNullable(customObject)
-                        .map(CustomObject::getValue)
-                        .map(node -> node.get(EMAIL_PROPERTY_STATUS))
-                        .map(JsonNode::asText)
-                        .filter(status -> ((equalsIgnoreCase(status, STATUS_PENDING) || tenantConfig.isProcessAll())))
-                        .map(pending -> callApiEndpoint(customObject.getId(), tenantConfig)
-                            .thenAccept(statistics::update))
-                        .orElseGet(() -> {
-                            statistics.update(Statistics.RESPONSE_IGNORED);
-                            return CompletableFuture.completedFuture(null);
-                        }))
-                    .collect(toList());
-
-                // join all the email processors
-                allOf(allTenants.toArray(new CompletableFuture[0])).join();
-
-                client.close();
-                return statistics;
-            })
-            .exceptionally(exception -> {
-                LOG.error(String.format("[%s] An unknown error occurred", tenantConfig.getProjectKey()),
-                    exception);
-                client.close();
-                return Statistics.ofError(tenantConfig.getProjectKey());
-            });
+        CustomObjectQuery<JsonNode> query = CustomObjectQuery.ofJsonNode().byContainer(CONTAINER_ID);
+        if (!tenantConfig.isProcessAll()) {
+            query = query.plusPredicates(QueryPredicate.of("value(status=\"" + STATUS_PENDING + "\")"));
+        }
+        query = query.withSort(s -> s.createdAt().sort().asc());
```

Comment:
Maybe add a comment why you sort ascendingly above?

Reply:
04e113b2c53a5165504e82ebf8290a58d132ec47